src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4863,30 +4863,30 @@ void PhaseIdealLoop::build_and_optimize() {
      C->set_major_progress();
   }
 
-  // Convert scalar to superword operations at the end of all loop opts.
+  // Auto-vectorize main-loop
   if (C->do_superword() && C->has_loops() && !C->major_progress()) {
     Compile::TracePhase tp("autoVectorize", &timers[_t_autoVectorize]);
-    // SuperWord transform
-    SuperWord sw(this);
+
+    // Shared data structures for all AutoVectorizations, to reduce allocations
+    // of large arrays.
+    VSharedData vshared;
```

Comment:
So it is local for each `build_and_optimize()` call and space will be freed by destructor.

Reply:
Yes, exactly.
Before my change, we had a SuperWord object per `build_and_optimize()` that allocated all the data structures.
So the scope is now still the same.
Except that before it all went to the `comp_arena`.

So before, we might have allocated those data structures multiple times (once per `build_and_optimize`), and grown `comp_arena` each time. Now we put it to a dedicated arena, which is freed in the destructor. So the memory usage should be a little lower that way.